tests/transforms/test_defer_measurements.py

```diff
@@ -335,6 +396,79 @@ def qnode():
 
         assert qnode() == expected
 
+    @pytest.mark.parametrize("device", ["default.qubit", "default.mixed", "lightning.qubit"])
+    def test_cond_qfunc(self, device):
+        """Test that a qfunc can also used with qml.cond."""
+        dev = qml.device(device, wires=2)
+
+        r = 2.324
+
+        @qml.qnode(dev)
+        def normal_circuit(rads):
+            qml.Hadamard(0)
+
+            qml.CNOT(wires=[0, 1])
+            qml.CRY(rads, wires=[0, 1])
+            qml.CZ(wires=[0, 1])
+            return qml.probs(wires=1)
+
+        def f(x):
+            qml.PauliX(1)
+            qml.RY(x, wires=1)
+            qml.PauliZ(1)
+
+        @qml.qnode(dev)
+        @qml.defer_measurements
+        def quantum_control_circuit(rads):
+            qml.Hadamard(0)
+            m_0 = qml.measure(0)
+            qml.cond(m_0, f)(r)
+            return qml.probs(wires=1)
+
+        exp = normal_circuit(r)
+        cond_probs = quantum_control_circuit(r)
+
+        assert np.allclose(exp, cond_probs)
```

Comment:
@antalszava I would highly suggest adding unit tests for this behaviour, probably in `test_condition`! To ensure that `qml.cond(m_0, f)(r)` is queuing the expected `Conditional` objects.

The reason being: while integration tests are good, they do not always provide the best coverage. E.g., it could be the case that `exp` and `cond_probs` are both trivial/zero!

Reply:
Great point! Added two unit tests in https://github.com/PennyLaneAI/pennylane/pull/2275/commits/58f531acfbc91e3ed50d84843f80ecec9244675d.